lib/elastic_apm/metrics/vm.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  module Metrics
+    # @api private
+    class VM
+      include Logging
+
+      SCOPE = 'runtime.ruby.'
+
+      # @api private
+      class Sampler
+        def sample
+          stat = GC.stat
+          total_time = GC::Profiler.total_time
+          thread_count = Thread.list.count
+
+          stat.merge(
+            gc_total_time: total_time,
+            thread_count: thread_count
+          )
+        end
+      end
+
+      def initialize(config)
+        @config = config
+        @sampler = Sampler.new
+      end
+
+      attr_reader :config, :sampler
+
+      def sample
+        @sampler.sample
+      end
+
+      def collect
+        return unless sampler
+
+        sample.each_with_object({}) do |(key, value), snap|
+          snap[:"#{SCOPE}#{key}"] = value
```

Comment:
I don't think we need to be capturing _all_ of the GC statistics, and we should be a bit more explicit about the names we choose to report. I suggest these names as a start:

 - ruby.gc.count
 - ruby.gc.time
 - ruby.thread.count (or ruby.threads? Java is using `.count`, but I'm not sure if that's the ideal convention)

Do you intend to add heap usage too?